supervisor.py

```diff
@@ -0,0 +1,158 @@
+        if tag == 'a':
+            for attr in attrs:
+                if attr[0] == 'href':
+                    if 'win32-static.zip' in attr[1]:
+                        self.static_download = attr[1]
+
+
+def check_for_git():
+    """
+    Does git exist? let's find out.
+    :return: True or False
+    """
+    try:
+        subprocess.check_output(['git', '--version'])
+        return True
+    except Exception as e:
+        return False
+
+
+def check_for_ffmpeg():
+    """
+    Does ffmpeg exist? let's find out
+    :return:  True or False
+    """
+    try:
+        subprocess.check_output(['ffmpeg', '-version'])
+        return True
+    except Exception as e:
+        return False
+
+
+class ProcessManager:
+    """
+    This class manages monitoring and restarting the process when it dies.
+    """
+    def __init__(self):
+        """
+        Initialize the class.
+        :ivar cmd: Holds the executable path for the current python interpreter running this supervisor. This will also be used to run the bot.
+        :ivar process: Holds the current subprocess popen reference
+        """
+        self.cmd = sys.executable
+        self.process = None
+
+    def start(self):
+        """
+        Start the subprocess and separate it to a new process group so they can run independently.
+        :return: True or False
+        """
+        if os.name == 'nt':
+            try:
+                self.process = subprocess.Popen([self.cmd, 'red.py'],
+                                                stdin=subprocess.DEVNULL,
+                                                stdout=subprocess.DEVNULL,
+                                                stderr=subprocess.STDOUT,
+                                                creationflags=subprocess.CREATE_NEW_PROCESS_GROUP)
+                return True
+            except Exception as e:
+                return False
+        else:
+            try:
+                self.process = subprocess.Popen([self.cmd, 'red.py'],
+                                                stdin=subprocess.DEVNULL,
+                                                stdout=subprocess.DEVNULL,
+                                                stderr=subprocess.STDOUT,
+                                                preexec_fn=os.setpgrp)
+                return True
+            except Exception as e:
+                return False
+
+    def wait(self):
+        self.process.wait()
+        return self.process.returncode
+
+
+if __name__ == "__main__":
+    # Check for Git
+    if not check_for_git():
+        print("Please install GIT to properly use this supervisor.")
+        sys.exit(0)
+
+    # Check for FFMPEG
+    if os.name == 'nt':
+        if not check_for_ffmpeg():
+            if IS_X8664:
+                print("FFMPEG not found. Downloading.")
+                with urllib.request.urlopen("https://github.com/Twentysix26/Red-DiscordBot/raw/master/ffmpeg.exe") as data:
```

Comment:
I didn't actually look to see if those links were the same besides the programs. Yeah, those could be a loop.